File: aNdroid11am/AdapterExamples/app/src/main/java/com/example/adapterexamples/CustomArrayAdapter.java
package com.example.adapterexamples;

import android.content.Context;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.ArrayAdapter;
import android.widget.ImageView;
import android.widget.TextView;

import androidx.annotation.NonNull;
import androidx.annotation.Nullable;

import org.w3c.dom.Text;

import java.util.ArrayList;

public class CustomArrayAdapter extends ArrayAdapter<ModelItem> {

    ArrayList<ModelItem> arrayList;
    Context context;
    public CustomArrayAdapter(@NonNull Context context, int resource,ArrayList<ModelItem> arrayList) {
        super(context, resource);
        this.arrayList = arrayList;
        this.context = context;

    }

    @Override
    public int getCount() {
        return arrayList.size();
    }

    @NonNull
    @Override
    public View getView(int position, @Nullable View convertView, @NonNull ViewGroup parent) {
        View view = convertView;
        LayoutInflater layoutInflater =(LayoutInflater) context.getSystemService(Context.LAYOUT_INFLATER_SERVICE);
       view = layoutInflater.inflate(R.layout.array_format,null);
        ImageView imageView = view.findViewById(R.id.format_imageview);
        TextView username = view.findViewById(R.id.format_username);

        imageView.setImageResource(arrayList.get(position).getImageurl());
        username.setText(arrayList.get(position).getName());

        return view;
    }
}

File: 4pmclass/TypesofAdapters/app/src/main/java/com/example/typesofadapters/apdatper/CustomArrayAdapter.kt
package com.example.typesofadapters.apdatper

import android.content.Context
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.ArrayAdapter
import android.widget.ImageView
import android.widget.TextView
import com.example.typesofadapters.R
import com.example.typesofadapters.modelclass.Item

class CustomArrayAdapter(var textviewresourcId:Int,var imagelist: MutableList<Item>,var contextloc:Context) :
    ArrayAdapter<Item>(contextloc,textviewresourcId,imagelist){

    override fun getCount(): Int {
        return imagelist.size
    }

    override fun getView(position: Int, convertView: View?, parent: ViewGroup): View {
        var view = convertView
        val inflater = contextloc.getSystemService(Context.LAYOUT_INFLATER_SERVICE) as LayoutInflater
        view = inflater.inflate(R.layout.arrayformat,null)

        var imageView = view.findViewById<ImageView>(R.id.imagelist)
        var textView = view.findViewById<TextView>(R.id.nametext)

        var item = imagelist.get(position)

        imageView.setImageResource(item.imageurl)
        textView.text = item.username


        return view
    }
}